addon/mixins/modal.js

```diff
@@ -25,25 +25,41 @@ export default Ember.Mixin.create(Base, {
 
   onHide: function() {
     this.set('hiding', true);
+  },
+
+  onHidden: function() {
     if (this.get('controller')) {
```

Comment:
During my tests controller isn't around anymore when onHidden hits which mean we never send the closeModal message. Have you checked this?